src/platform/plugins/shared/discover/public/application/main/components/top_nav/app_menu_actions/get_share.tsx

```diff
@@ -114,7 +114,17 @@ export const getShareAppMenuItem = ({
           },
           sharingData: {
             isTextBased: isEsqlMode,
-            locatorParams: [{ id: locator.id, params }],
+            locatorParams: [
+              {
+                id: locator.id,
+                params: isEsqlMode
+                  ? {
+                      ...params,
+                      timeRange: timefilter.getAbsoluteTime(), // Will be used when generating CSV on server. See `filtersFromLocator`.
```

Comment:
I guess this isn't an issue for classic mode because we don't use csv_v2 currently? If it's not used for classic mode atm, I wonder if there would be drawbacks to just always passing this regardless of mode? Just thinking it would prepare us if we ever migrated classic mode to csv_v2.

Reply:
It can prepare or it can break current things 😀 Didn't want to change anything besides what is necessary to address the bug. We could revisit it later if becomes required.

Reply:
That's fair, definitely don't want to add risk to this bug fix 🙂 We can update it later if we migrate classic mode to csv_v2.